src/web/routes/application/select-address/select-address.js

```diff
@@ -17,22 +19,33 @@ const buildAddressOption = result => ({
   text: result.ADDRESS
 })
 
-const behaviourForGet = () => async (req, res, next) => {
+const behaviourForGet = () => (req, res, next) => {
   res.locals.addresses = req.session.postcodeLookupResults.map(buildAddressOption)
   // Manual address is further in the flow than select-address, therefore this line is needed to prevent the state machine from redirecting the user back to select-address.
   req.session.nextAllowedStep = '/manual-address'
   next()
 }
 
+const behaviourForPost = () => (req, res, next) => {
+  const address = req.session.postcodeLookupResults.find(result => result.ADDRESS === req.body.selectedAddress)
```

Comment:
I think this would be clearer as its own function.

Reply:
Possibly should explicitly deal with the situation that the address is not found with try / catch (not sure how this would happen)? Would we want to log that as an error?

Reply:
I've extracted this logic to it's own function which throws an error if no address is found.